### ISSUE TITLE
Develop WDL specification further (second specification)

### ISSUE THREAD

**Issue by thoranderskow** [uchicago-cs/chiventure#55]
Edit existing WDL file format to include more specific examples of file objects. Discuss with Game_State to see which file objects should be added/modified.

**Comment by thoranderskow** [comment]
Specification modifications that we decided must be implemented:

-A list of actions (requires Action_management coordination)
-A Player file object(may require game_state coordination)
-Extensive examples (like prototypes of whole file objects)
-Explain ambiguous language and keywords

**Comment by bansals10** [comment]
Access list of actions under src/action_management/include/actionmanagement.h on the actions/dev branch on github

**Comment by thoranderskow** [comment]
List of supported actions from AM branch 
OPEN, /* world action */
  CLOSE, /* world action */
  PUSH, /* world action */
  PULL, /* world action */
  EXAMINE, /* synonymous to "look at", is a world action */
  TURN_ON, /* world action */
  TURN_OFF, /* world action */
  TALK_TO, /* NPC action */
  GIVE, /* requires 2 arguments separated by 'to',  NPC action */
  GO, /* world action */
  TAKE, /* synonymous with "pick up", is an inventory action */
  DROP, /* inventory action */
  CONSUME, /* synonymous with "use", "drink", "eat", is an inventory action */
  USE_ON, /* requires two arguments separated by 'on', world and NPC action  */

**Comment by thoranderskow** [comment]
Example of in-depth example of file object

ROOMS:
 - id: KITCHEN
   short_desc: "A well-furnished area for cooking food."
   long_desc: "The room	is lit merrily,	and a faint glow comes from the	oven."
   connections:
    - to: BASEMENT
      direction: DOWN
      through: trapdoor


EXAMPLE:
<FILE OBJECT CATEGORY>
 - id: <UNIQUE ID NAME>
   short_desc: <STRING DESCRIPTION>
   long_desc: <STRING DESCRIPTION>
   connections:
    - to: <ROOM ID>
      direction: <CARDINAL DIRECTION>
      through: <ITEM ID>

**Comment by bansals10** [comment]
I moved the WDL_README_DRAFT2.md from src to docs in the wdl/dev branch.

**Comment by Kate32815** [comment]
We added definitions for common phrases used, such as "attribute" and "object", created syntax specifications, and further explained terms to reduce ambiguity. We also added full examples for the game, room, items objects.

For actions, we decided to put in a link to Action Management's specifications rather than restating them in the specification. For the potential Player object, we decided it was not necessary to specify it in a WDL file given the current state of the game with only one player.

We may create an NPC (Non-Player Character) object in the future if there is interest from the other teams to create NPC interactions. I am closing the issue.

**Comment by Kate32815** [comment]
We added definitions for common phrases used, such as "attribute" and "object", created syntax specifications, and further explained terms to reduce ambiguity. We also added full examples for the game, room, items objects.

For actions, we decided to put in a link to Action Management's specifications rather than restating them in the specification. For the potential Player object, we decided it was not necessary to specify it in a WDL file given the current state of the game with only one player.

We may create an NPC (Non-Player Character) object in the future if there is interest from the other teams to create NPC interactions. I am closing the issue.
